gemrb/plugins/GUIScript/GUIScript.cpp

```diff
@@ -7422,32 +7462,37 @@ the PC's inventory.\n\
 static PyObject* GemRB_ChangeStoreItem(PyObject * /*self*/, PyObject* args)
 {
 	int globalID, Slot;
-	int action;
+	int actint;
 	int res = ASI_FAILED;
-	PARSE_ARGS( args,  "iii", &globalID, &Slot, &action);
+	PARSE_ARGS( args,  "iii", &globalID, &Slot, &actint);
 	GET_GAME();
 	GET_ACTOR_GLOBAL();
 
 	Store *store = core->GetCurrentStore();
 	if (!store) {
 		return RuntimeError("No current store!");
 	}
+
+	StoreActionFlags action = static_cast<StoreActionFlags>(actint);
+	
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wswitch"
```

Comment:
Is this still needed now that it has a default clause?

Reply:
apparently still required. The problem is the bitwise operations yielding values not defined in the` enum`:

```
[ 66%] Building CXX object gemrb/plugins/GUIScript/CMakeFiles/GUIScript.dir/GUIScript.cpp.o
/home/runner/work/gemrb/gemrb/gemrb/plugins/GUIScript/GUIScript.cpp: In function ‘PyObject* GemRB_ChangeStoreItem(PyObject*, PyObject*)’:
/home/runner/work/gemrb/gemrb/gemrb/plugins/GUIScript/GUIScript.cpp:7548:2: error: case value ‘65’ not in enumerated type ‘GemRB::StoreActionFlags’ [-Werror=switch]
  case StoreActionFlags::Select | StoreActionFlags::Buy:
  ^~~~
/home/runner/work/gemrb/gemrb/gemrb/plugins/GUIScript/GUIScript.cpp:7564:2: error: case value ‘66’ not in enumerated type ‘GemRB::StoreActionFlags’ [-Werror=switch]
  case StoreActionFlags::Select | StoreActionFlags::Sell:
  ^~~~
/home/runner/work/gemrb/gemrb/gemrb/plugins/GUIScript/GUIScript.cpp:7565:2: error: case value ‘68’ not in enumerated type ‘GemRB::StoreActionFlags’ [-Werror=switch]
  case StoreActionFlags::Select | StoreActionFlags::ID:
  ^~~~
cc1plus: all warnings being treated as errors
```